File: standardnaast-backend/src/main/java/standardNaast/backend/service/TeamService.java
package standardNaast.backend.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import standardNaast.backend.dto.TeamCreateUpdateDto;
import standardNaast.backend.dto.TeamDto;

import java.util.List;

public interface TeamService {

    Page<TeamDto> getTeams(String search, Pageable pageable);

    List<TeamDto> getAllTeams();

    TeamDto getTeamById(Long id);

    TeamDto createTeam(TeamCreateUpdateDto dto);

    TeamDto updateTeam(Long id, TeamCreateUpdateDto dto);

    void deleteTeam(Long id);
}

File: standardnaast-ng/src/app/core/services/team.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Team, TeamCreateUpdate } from '../models/team.model';
import { Page } from '../models/page.model';

@Injectable({
  providedIn: 'root'
})
export class TeamService {
  private readonly apiUrl = `${environment.apiUrl}/teams`;

  constructor(private http: HttpClient) {}

  getTeams(search?: string, page = 0, size = 20, sort = 'name,asc'): Observable<Page<Team>> {
    let params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString())
      .set('sort', sort);

    if (search && search.trim()) {
      params = params.set('search', search.trim());
    }

    return this.http.get<Page<Team>>(this.apiUrl, { params });
  }

  getAllTeams(): Observable<Team[]> {
    return this.http.get<Team[]>(`${this.apiUrl}/all`);
  }

  getTeamById(id: number): Observable<Team> {
    return this.http.get<Team>(`${this.apiUrl}/${id}`);
  }

  createTeam(team: TeamCreateUpdate): Observable<Team> {
    return this.http.post<Team>(this.apiUrl, team);
  }

  updateTeam(id: number, team: TeamCreateUpdate): Observable<Team> {
    return this.http.put<Team>(`${this.apiUrl}/${id}`, team);
  }

  deleteTeam(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
